### ISSUE TITLE
Show richer search results

### ISSUE THREAD

**Issue by tsmacdonald** [metabase/metabase#14722]
Once #14583 lands the backend will be returning richer match information, and now we need to show it on the frontend.

This will certainly involve showing the contents of `matched_text` if it's not a `name`. It's also a good opportunity to audit where we display `name` and where we display `display_name`.

We should also bold the actual match and truncate the `matched_text` if its too long, but that might get spun off into its own issue

**Comment by tsmacdonald** [comment]
Fixed by #14867
